toolsrc/src/vcpkg/commands.ci.cpp

```diff
@@ -436,36 +461,16 @@ namespace vcpkg::Commands::CI
                 serialize_options.randomizer = &randomizer_instance;
             }
 
-            auto action_plan = [&]() {
-                int iterations = 0;
-                do
-                {
-                    bool inconsistent = false;
-                    auto action_plan = pgraph.serialize(serialize_options);
-
-                    for (auto&& action : action_plan)
-                    {
-                        if (auto p = action.install_action.get())
-                        {
-                            p->build_options = install_plan_options;
-                            if (Util::Sets::contains(exclusions_set, p->spec.name()))
-                            {
-                                p->plan_type = InstallPlanType::EXCLUDED;
-                            }
-
-                            for (auto&& feature : split_specs->features[p->spec])
-                                if (p->feature_list.find(feature) == p->feature_list.end())
-                                {
-                                    pgraph.install({p->spec, feature});
-                                    inconsistent = true;
-                                }
-                        }
-                    }
+            auto action_plan = Dependencies::PackageGraph::create_feature_install_plan(
```

Comment:
This needs to be iterative (similar to previous implementation) in order to account for non-default features being required in CI.

Specific circumstance:
1. `pango -> harfbuzz[glib]`
2. `foo -> harfbuzz[core]`

`pango` and `harfbuzz` have known results, `foo` does not. `foo` is the only element of `split_specs->unknown`, which generates a graph containing `foo[core]` and `harfbuzz[core]`. This will not build the originally requested foo package (which should be built against `harfbuzz[glib]`) and will cause the CI to rebuild foo on every run.

Reply:
Second action plan is created with provider with modified default features.